044-DevOpswithGithubActions/Coach/README.md

```diff
@@ -0,0 +1,32 @@
+# Welcome to Devops With GitHub Hackathon
+
+## Overview
+
+Contoso is Insurance company, they provide a full range of long-term insurance services to help individuals who are under-insured, filling a void their founders saw in the market. From the beginning, they grew faster than anticipated and have struggled to cope with rapid growth. During their first year alone, they added over 100 new employees to keep up with the demand for their services. To manage policies and associated documentation, they use a custom-developed Windows Forms application, called Policy Connect. Policy Connect uses an Azure Sql database as its data store. 
+Contoso recently started a new web and mobile projects to allow policyholders, brokers, and employees to access policy information without requiring a VPN connection into the Contoso network. The web project is a new .NET Core 3.1 MVC web application, which accesses the PolicyConnect database using REST APIs. They eventually intend to share the REST APIs across all their applications including the mobile app. They already have plan in place to Host Web Applications on Azure. Contoso Architecture team has already created blueprint of their deployments on Azure. Work with your team to Assess Solution requirements. Create step by step plan as listed in Hackathon tasks and deploy solution components. Code repository is hosted on Github. Contoso team needs your help to create Github Workflows to automate CI/CD and deploy application on Azure. Security is prime concern for them hence ensure secrets are stored in secured vaults. Policy Documents for Policy Holders are stored in Cloud Storage and retrieved using Web and Mobile apps using APIs.
+
+
+![Solution BluePrint](/044-DevOpswithGithubActions/Student/Resources/images/solutionblueprint.png)
+
+## Coach Guidance for Challenges
+- Challenge 1: **[Assessment](./Challenge01.md)**
+   - Navigate through Source code and services required
+   - Create service principal with contributer role on Azure Subscription
+ - Challenge 2: **[Service creation](./Challenge02.md)**
+   - Create Azure services for SQL Databases and API Website .
+   - Build and Deploy API website.
+- Challenge 3: **[Deploy Front end Website](./Challenge03.md)**
+   - Create Azure Services for Front Website.
+   - Build and Deploy Frontend Website
+- Challenge 4: **[Power of Serverless](./Challenge04.md)**
+   - Create Azure Function Website.
+   - Build and Deploy Azure Function Wesbite.
+- Challenge 5: **[Secure your Solution](./Challenge05.md)**
+   - Create API Management service.
+   - Import Open API and Azure Function API.
+
+   
+## Prerequisites
+
```

Comment:
There should be additional information included here in the Prerequisites section for coaches.  

Please list out the Azure resources a student will consume if implementing all of the solutions for this hack.

This is important for a potential customer to know if deciding to host the hack in their own Azure subscription.  For example, list out:

"Each student will consume the following resources in Azure to complete this hack:
-1 2-core VM
-3 Azure Functions
-??? Storage
-3 public IP addresses
-etc

If students are using a shared subscription, ensure quotas are raised so that provisioning errors don't occur"

Also, what permissions are needed by the student (and or the coachs) to set up the environment?  Owner?  Contributor? Etc?    These are common issues that trip up coaches, students, and the stakeholder at the organization where the hack is being delivered.  Calling them out here make it easier for all.